src/hotspot/share/opto/escape.cpp

```diff
@@ -527,41 +534,52 @@ bool ConnectionGraph::has_been_reduced(PhiNode* n, SafePointNode* sfpt) const {
 //  - Phi -> AddP -> Load
 //  - Phi -> CastPP -> SafePoints
 //  - Phi -> CastPP -> AddP -> Load
-bool ConnectionGraph::can_reduce_check_users(Node* n, uint nesting) const {
+//  - Phi -> Phi -> AddP -> Load
+bool ConnectionGraph::can_reduce_check_users(Node* n, uint phi_nest_level) const {
   for (DUIterator_Fast imax, i = n->fast_outs(imax); i < imax; i++) {
     Node* use = n->fast_out(i);
-
-    if (use->is_SafePoint()) {
+    // Skip Phi -> Phi -> SafePoints and allow only Phi -> SafePoints, Phi -> CastPP -> SafePoints, 
```

Comment:
"// Skip Phi -> Phi -> SafePoints and allow only Phi -> SafePoints and Phi -> CastPP -> SafePoints"